### PR TITLE
Setup or manage required Hadoop related OS groups

### PR DESCRIPTION
**Pull request by amithkanand**
This PR takes are of adding or managing Hadoop related OS groups.

### CODE REVIEW

File: cookbooks/bcpc-hadoop/libraries/utils.rb [hunk 1]
```diff
@@ -378,3 +378,21 @@ def user_exists?(user_name)
   return user_found
 end
 
+def group_exists?(group_name)
+  group_found = false
+  chk_grp_cmd = "getent group #{group_name}"
+  Chef::Log.debug("Executing command: #{chk_grp_cmd}")
+  cmd = Mixlib::ShellOut.new(chk_grp_cmd, :timeout => 10).run_command
+  if cmd.exitstatus == 0
```
Comment (pu239ppy) [inline cookbooks/bcpc-hadoop/libraries/utils.rb]: Just a style suggestion
return cmd.exitstatus == 0

**Comment by pu239ppy** [comment]
Looks good to me

**Comment by cbaenziger** [comment]
I believe this could cause issues on systems which end up getting a SSO configuration in their converge. One would use a `:create` action in that case but when the `group` resource runs the group would already exist and fail out since the group information is not local to the machine?

**Comment by amithkanand** [comment]
@cbaenziger: I changed the code to use a `Ruby Block` and that should take care of your concern.

**Comment by cbaenziger** [comment]
This looks good to me; a couple of formatting nits which [rubocop](http://batsov.com/rubocop/) would gripe on but we need this in master for now so I'm happy to merge.

File: cookbooks/bcpc-hadoop/libraries/utils.rb [hunk 2]
```diff
@@ -378,3 +378,12 @@ def user_exists?(user_name)
...
 
+def group_exists?(group_name)
+  chk_grp_cmd = "getent group #{group_name}"
+  Chef::Log.debug("Executing command: #{chk_grp_cmd}")
+  cmd = Mixlib::ShellOut.new(chk_grp_cmd, :timeout => 10).run_command
+  return  cmd.exitstatus == 0 ? true : false 
+end
+
+def get_group_action(group_name)
+  return  group_exists?(group_name) ? :manage : :create
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/libraries/utils.rb]: A trailing end would be useful here.

Comment (amithkanand) [inline cookbooks/bcpc-hadoop/libraries/utils.rb]: Trailing `end` is there. Not sure why not visible here.

**Pull request merged**